analysis/quantifier-comparisons/10-miQC-filtering.Rmd

```diff
@@ -0,0 +1,234 @@
+# test polynomial mixture model rather than linear
+purrr::iwalk(cr_like_sce, model_sce, model_type = "polynomial")
+```
+
+For the linear model, we still see that cells with low mito are getting excluded from SCPCR000118 and SCPCR000119, but now see that same trend in SCPCR000220 and SCPCR000221 as well. 
+It appears to model the cells with very high number of unique genes and low mito content as cells to exclude when we would rather keep those. 
+
+## Filter Cells
+
+Now we can compare filtering using miQC vs. using the pre-determined cutoffs to see how the different filtering effects the overall number of cells that are removed. 
+We can also look at the distribution of each of the metrics used for filtering, genes detected/cell, and mito content/cell to see how filtering affects the population of cells. 
+Here we are using a threshold of 500 genes detected per cell and 10% mitochondrial reads.
+
+```{r}
+# function to filter sce using miQC 
+miQC_filter <- function(sce, model_type = "linear", posterior_cutoff = 0.75){
+  sce_model <- miQC::mixtureModel(sce, model_type)
+  filtered_sce <- miQC::filterCells(sce, sce_model)
+}
+
+# filter using manual 
+manual_filter <- function(sce, 
+                          mito_cutoff = 10, 
+                          genes_detected_cutoff = 500) {
+  
+  # get vector of cells to keep 
+  cells_keep <- sce$detected > genes_detected_cutoff & 
+    sce$subsets_mito_percent < mito_cutoff
+  
+  # filter sce
+  filtered_sce <- sce[,cells_keep]
+}
+
+manual_filtered_sce <- purrr::map(cr_like_sce, manual_filter)
+miQC_filtered_sce <- purrr::map(cr_like_sce, miQC_filter)
+```
+
+
+```{r}
+# function to grab number of cells from sce
+get_num_cells <- function(sce){
+  num_cells <- dim(sce)[2]
+}
+
+manual_cell_num <- manual_filtered_sce %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(manual_cell_num) <- c("manual")
+
+miQC_cell_num <- miQC_filtered_sce %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(miQC_cell_num) <- c("miQC")
+
+pre_filter_cell_num <- cr_like_sce %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(pre_filter_cell_num) <- c("pre_filtering")
+
+# create dataframe for plotting with number of cells per sce 
+combined_cell_num_df <- dplyr::bind_rows(manual_cell_num,
+                                         miQC_cell_num,
+                                         pre_filter_cell_num) %>%
+  tibble::rownames_to_column("filtering_method") %>%
+  tidyr::pivot_longer(cols = starts_with("SCPCR"),
+                      names_to = "sample", 
+                      values_to = "number_cells")
+```
+
+```{r}
+ggplot(combined_cell_num_df, aes(x = sample, y = number_cells, color = filtering_method)) + 
+  geom_point(size = 3) + 
+  theme(axis.text.x = element_text(angle = 90))
+```
+
+
+It looks like manual filtering is removing more cells than miQC for single-cell, but similar cell numbers are removed in single-nuclei samples with either manual or miQC. 
```

Comment:
Doesn't this just depend on the cutoffs we are using though? If we were using 20%, we would be getting a different relationship.